api/src/main/java/org/openmrs/module/muzima/api/db/hibernate/HibernateDataDao.java

```diff
@@ -187,22 +186,29 @@ public void purgeData(final T data) {
             disjunction.add(Restrictions.ilike("formName", search, MatchMode.ANYWHERE));
             disjunction.add(Restrictions.ilike("provider.identifier", search, MatchMode.ANYWHERE));
             disjunction.add(Restrictions.ilike("provider.name", search, MatchMode.ANYWHERE));
-            if(StringUtils.isNumeric(search)) {
-                disjunction.add(Restrictions.eq("location.locationId", Integer.parseInt(search)));
-            }
-            criteria.add(disjunction);
-
-        }
-        if (pageNumber != null) {
-            criteria.setFirstResult((pageNumber - 1) * pageSize);
+	        if(StringUtils.isNumeric(search)) {
+		        disjunction.add(Restrictions.eq("location.locationId", Integer.parseInt(search)));
+	        }
+	        if(errorIds.size() > 0) {
+		        disjunction.add(Restrictions.in("id", errorIds));
```

Comment:
Kindly review this. You are comparing ErrorData.id against ErrorMessage.id, which are not equivalent.